test/test_base_service.py

```diff
@@ -553,6 +553,27 @@ def test_gzip_compression_external():
     assert prepped['data'] == gzip.compress(b'{"foo": "bar"}')
     assert prepped['headers'].get('content-encoding') == 'gzip'
 
+@responses.activate
+def test_retry_config_external():
```

Comment:
I think we need several different tests to put the retries support through its paces.  For examples of the types of tests you could look at `base_service_retries_test.go` in the Go core.